src/Microplate.php

```diff
@@ -0,0 +1,198 @@
+<?php declare(strict_types=1);
+
+namespace Mll\Microplate;
+
+use Illuminate\Support\Collection;
+use Mll\Microplate\Enums\FlowDirection;
+use Mll\Microplate\Exceptions\MicroplateIsFullException;
+use Mll\Microplate\Exceptions\UnexpectedFlowDirection;
+use Mll\Microplate\Exceptions\WellNotEmptyException;
+
+/**
+ * @template TWell
+ *
+ * @phpstan-type WellsCollection Collection<array<string, TWell>>
+ */
+class Microplate
+{
+    public const EMPTY_WELL = null;
+
+    public CoordinateSystem $coordinateSystem;
+
+    /**
+     * @var WellsCollection
```

Comment:
Inaccurate

```suggestion
     * @var Collection<string, TWell>
```

Perhaps this project needs Larastan in order to have proper generic support for the `Collection` class

Reply:
I tried to get it running. There are some issues installing it without Laravel are commented here:
https://github.com/nunomaduro/larastan/issues/559
I'd suggest to find a solution for it in a new ticket/branch

Reply:
composer require --dev laravel/framework

This should do the trick